### PR TITLE
Feature/actor lines chart

### PR DESCRIPTION
**Pull request by pjosh**
captura de pantalla 2017-05-04 a las 9 30 50 2

### CODE REVIEW

**Comment by pjosh** [comment]
Fixes #321

File: public/mocks/v1_get_actor_node_attributes.json [hunk 1]
```diff
@@ -11,190 +11,223 @@
...
+              47000,
+              50000,
+              70000,
+              47000
             ]
           },
           {
             "id": 1915,
-            "name": "NOVA UBIRATA",
+            "name": "Sidrolândia",
```
Comment (nerik) [inline public/mocks/v1_get_actor_node_attributes.json:214]: Check with Agnieszka, but I think the DB only has non accentuated names in caps, so we'll have to deal this in the frontend anyways.

File: scripts/pages/profile-actor.page.js [hunk 2]
```diff
@@ -16,14 +16,16 @@ import 'styles/components/profiles/error.scss';
 import Nav from 'components/shared/nav.component.js';
 import Dropdown from 'components/shared/dropdown.component';
 import Map from 'components/profiles/map.component';
-import Top from 'components/profiles/top.component';
+//import Top from 'components/profiles/top.component';
```
Comment (nerik) [inline scripts/pages/profile-actor.page.js]: Remove

Comment (nerik) [reply scripts/pages/profile-actor.page.js]: (and remove component if it's not used anymore)

File: scripts/pages/profile-actor.page.js [hunk 3]
```diff
@@ -41,12 +43,36 @@ const _build = (data, nodeId) => {
   const tooltip = new Tooltip('.js-infowindow');
 
   if (data.top_sources.municipalities.lines.length) {
-    new Top({
-      el: document.querySelector('.js-top-municipalities'),
-      data: data.top_sources.municipalities.lines,
-      targetLink: 'place',
-      title: 'top source municipalities in 2015'
-    });
+    document.querySelector('.js-top-municipalities-title').innerHTML = `Top source regions of ${formatApostrophe(_.capitalize(data.node_name))} soy:`;
```
Comment (nerik) [inline scripts/pages/profile-actor.page.js]: We'll have to implement being able to switch between municipalities/states/biomes eventually, but in the meantime I would put 'Top source regions of Brazil's soy: municipalities'

File: scripts/components/profiles/line.component.js [hunk 4]
```diff
@@ -36,87 +39,142 @@ export default class {
       .append('g')
       .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
 
+    let tempArr = data.lines.map(line => line.values);
+    let allYValues = [];
+    for(let i = 0; i < tempArr.length; i++) {
+      allYValues = [...allYValues, ...tempArr[i]];
+    }
```
Comment (nerik) [inline scripts/components/profiles/line.component.js]: Couldn't this block be replaced by `[].concat(data.lines.map(line => line.values))` ?

File: scripts/pages/profile-actor.page.js [hunk 5]
```diff
@@ -71,11 +97,36 @@ const _build = (data, nodeId) => {
...
+        showTooltipCallback: (country, x, y) => {
+          tooltip.showTooltip(x, y, {
+            title: `${data.node_name} > ${country.name.toUpperCase()}, ${country.date.getFullYear()}`,
+            values: [
+              { title: 'Trade Volume',
+                value: `${formatNumber(country.value)}<span>Tons</span>` }
+            ]
+          });
+        },
+        hideTooltipCallback: () => {
```
Comment (nerik) [inline scripts/pages/profile-actor.page.js]: Could the configuration of the Line component be made more DRY ? (ticks, showTooltipCallback, hideTooltipCallback look like they are the same from the other Line component)

File: scripts/pages/profile-actor.page.js [hunk 6]
```diff
@@ -71,11 +97,36 @@ const _build = (data, nodeId) => {
   }
 
   if (data.top_countries.lines.length) {
-    new Top({
-      el:document.querySelector('.js-top-destination'),
-      data: data.top_countries.lines,
-      title: 'top destination countries in 2015'
-    });
+    document.querySelector('.js-top-map-title').innerHTML = `TOP DESTINATION COUNTRIES Of ${formatApostrophe(_.capitalize(data.node_name))} SOY`;
```
Comment (nerik) [inline scripts/pages/profile-actor.page.js]: Capitalize should be doable with CSS, right?

**Review by nerik — COMMENTED** [review]
Nice work! I think factorizing line chart from both pages is a good move. And the result is 👍

**Comment by nerik** [comment]
Don't forget the changelog please!

File: scripts/components/profiles/line.component.js (hunk 4 above)
Comment (pjosh) [reply scripts/components/profiles/line.component.js]: In this case, `[].concat(data.lines.map(line => line.values))` return the same as `data.lines.map(line => line.values)`.
But you have helped me to do this `[].concat.apply([], data.lines.map(line => line.values))` 😄

File: public/mocks/v1_get_actor_node_attributes.json (hunk 1 above)
Comment (pjosh) [reply public/mocks/v1_get_actor_node_attributes.json:214]: What have you thought? In a dictionary of replacements?

**Comment by pjosh** [comment]
Thx for the amazing feedback @nerik !
All done!

**Pull request merged**